DashAI/back/models/hugging_face/vit_transformer.py

```diff
@@ -0,0 +1,164 @@
+import json
+import shutil
+
+import numpy as np
+from sklearn.exceptions import NotFittedError
+from transformers import (
+    Trainer,
+    TrainingArguments,
+    ViTFeatureExtractor,
+    ViTForImageClassification,
+)
+
+from DashAI.back.dataloaders.classes.dashai_dataset import DashAIDataset
+from DashAI.back.models.base_model import BaseModel
+from DashAI.back.models.image_classification_model import ImageClassificationModel
+
+
+class ViTTransformer(BaseModel, ImageClassificationModel):
+    """
+    Pre-trained transformer ViT allowing image classification
+    """
+
+    @classmethod
+    def get_schema(cls):
+        with open("DashAI/back/models/parameters/models_schemas/ViT.json") as f:
+            cls.SCHEMA = json.load(f)
+        return cls.SCHEMA
+
+    def __init__(self, model=None, **kwargs):
```

Comment:
If we follow the scikit-learn convention, all parameters received by this model should be explicitly defined. Then, you could group them in training_args dict.

This also gives you the possibility to verify them individually.

```suggestion
    def __init__(self, model=None, epochs: int, batch_size: int, learning_rate: float, device: Literal["gpu", "cpu"], weight_decay: float):
```


I also take this opportunity to ask, in which case model could be used 

Finally, please add a brief description to the parameters in the documentation.

Reply:
In this I have a couple of doubts, because I feel that it is not very extensible if we then add more parameters for the models (because there are many more).  model is used when the model is loaded, this way a new instance of a transformer is initialized but with the new weights.

Reply:
Ok, we will leave the topic for later.